inst/tests/tests.Rraw

```diff
@@ -18427,6 +18427,10 @@ test(2252.1, dt[, b:=2L], error = "\\[ was called on a data.table.*not data.tabl
 test(2252.2, dt[, let(b=2L)], error = "\\[ was called on a data.table.*not data.table-aware.*'let'")
 rm(.datatable.aware)
 
+lc_ctype = Sys.getlocale('LC_CTYPE')
+lc_collate = Sys.getlocale('LC_COLLATE')
+suppressWarnings(Sys.setlocale('LC_COLLATE', "en_US.UTF-8"))
```

Comment:
what's `suppressWarnings()` for here?

also, is `en_US.UTF-8` needed here? What about `C` instead? In which case we can re-use the `with_c_collate()` helper defined above...

Reply:
I used `suppressWarnings()` because another test in the suite did as well when setting locale, test 2011, but looking at it now `setlocale()` has been called a few times without `suppressWarnings()` so I assume it can be omitted.

As for `C`, I'll give this a try, before this I just used `UTF-8` as I just assumed it was the locale we needed as it contains the most symbols, could be wrong

Reply:
After some testing, it seems that these tests require `LC_CTYPE="en_US.UTF-8"`, and collate doesn't actually matter. The tests do fail with `LC_CTYPE=C`

Reply:
Thanks for confirming. Let's put them in a `local({...})` block to reset with `on.exit()`, that way `LC_CTYPE` is not left altered if the suite fails.